### PR TITLE
Key 167 | Docs accessibility items

### PR DESCRIPTION
**Pull request by ChrisLaneAU**
Fixes various accessibility items raised by axe and adds some screen reader UX.

There's still some outstanding items which I'll add to a new ticket.

Also fixed a duplicate aria-current on the mobile menu.

### CODE REVIEW

File: docs/src/app/(public)/page.tsx [hunk 1]
```diff
@@ -5,18 +5,19 @@ import MailingList from '../../components/mailing-list';
 import CallToAction from '../../components/call-to-action';
 import Footer from '../../components/footer';
 import Features from '../../components/features';
+import { Main } from '../../components/main';
 
 export default function Homepage() {
   return (
     <>
-      <main>
+      <Main isFocusable={false}>
```
Comment (ChrisLaneAU) [inline docs/src/app/(public)/page.tsx:13]: I'm not sure why but this is the only page that scrolls to the `<main>` on client side nav so setting `isFocusable` to `false` prevents adding `tabindex="-1"` and fixes the issue. The `scroll={false}` prop on `next/link` doesn't work ATM but maybe that's a better fix later on.

File: docs/src/components/navigation/mobile-nav.tsx
Comment (ChrisLaneAU) [inline docs/src/components/navigation/mobile-nav.tsx:1]: Mainly the changes here were for valid children of the `ul` and valid parents of some of the `li`. I believe it's still as you had designed @jtbrolin but let me know if I've missed anything.

File: docs/src/components/main.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+>;
+
+export const Main = ({
+  children,
+  className,
+  isFocusable = true,
+  ...props
+}: MainProps) => (
+  <main
+    className={`outline-0 ${className || ''}`}
```
Comment (BarnabyBishop) [inline docs/src/components/main.tsx:20]: I'm still getting an outline on ios:

image

Looks like it is a safari issue 😩
CleanShot 2023-07-07 at 10 38 42@2x

Comment (ChrisLaneAU) [reply docs/src/components/main.tsx:20]: Ahh bummer, thanks for spotting this! I'll take a look.

Comment (ChrisLaneAU) [reply docs/src/components/main.tsx]: Switching from `outline-0` to `outline-none` seems to have fixed it.

File: docs/src/components/features.tsx [hunk 3]
```diff
@@ -26,7 +26,11 @@ export default function Features() {
         </svg>
       </div>
 
-      <ul className="px-6 py-8 overflow-x-auto flex gap-4 md:gap-8" role="list">
+      <ul
+        tabIndex={0}
```
Comment (ChrisLaneAU) [inline docs/src/components/features.tsx:30]: Allows keyboard access for horizontally scrollable list.

File: docs/src/components/navigation/mobile-nav.tsx [hunk 4]
```diff
@@ -119,56 +121,56 @@ export function MobileNav({ navigationMap }: NavProps) {
...
+                href="/"
+                tabIndex={navOpen ? 0 : -1}
+              />
+
+              <NavItem
+                level="top"
+                label="Docs"
+                href="/docs"
+                tabIndex={navOpen ? 0 : -1}
+                currentPage={pathname === '/docs'}
```
Comment (ChrisLaneAU) [inline docs/src/components/navigation/mobile-nav.tsx:138]: @jtbrolin this changes the design a little bit but also prevents the duplicate `aria-current="page"`.

i.e. we go from:
image


To this:
image


Let me know if that's ok (to me it makes more sense but is inconsistent with desktop view).

Comment (jtbrolin) [reply docs/src/components/navigation/mobile-nav.tsx:138]: Yeah OK with me (I didn't notice the duplicate `aria-current="page"`, was too focused on the desktop<->mobile parity 😅)

Comment (ChrisLaneAU) [reply docs/src/components/navigation/mobile-nav.tsx:138]: Lol all good! I didn't notice until a follow-up look at something else this morning!

**Review by BarnabyBishop — APPROVED** [review]
Excellent updates 👍

**Pull request merged**